File: main/src/server/cmd/log.py
from queue import Queue

import zope
from loguru import logger

from server.interfaces.cmd import ICommand


@zope.interface.implementer(ICommand)
class CommandLog:
    """
    Реализация команды логирования посредствам решения loguru.
    """

    def __init__(self, exception: Exception):
        self.exception = exception

    def execute(self):
        logger.warning(self.exception)


def handler_log(exception: Exception, cmd_q: Queue):
    """Обработчик для добавления исключения в виде команды в очередь"""

    new_command = CommandLog(exception)
    cmd_q.put(new_command)

File: additional/internal/services/cmd/log/log.go
// Пакет отвечает за реализацию команды-логера исключений.
package log

import (
	"additional/internal/services/cmd"

	"github.com/sirupsen/logrus"
)

type Log struct {
	Exception error
}

func NewLog(exception error) *Log {
	return &Log{
		Exception: exception,
	}
}

func (cl Log) Execute() error {
	logrus.WithFields(logrus.Fields{"command": "log"}).Warn(cl.Exception)
	return nil
}

func HandlerLog(exception error, queue *cmd.CommandQueue[cmd.ICommand]) {
	newCommand := NewLog(exception)
	queue.Add(newCommand)
}
